data/src/main/java/com/microsoft/azure/kusto/data/Utils.java

```diff
@@ -118,18 +116,18 @@ static InputStream postToStreamingOutput(String url, String payload, long timeou
                 }
                 // Though the server responds with a gzip/deflate Content-Encoding header, we reach here because httpclient uses LazyDecompressingStream which handles the above logic
                 returnInputStream = true;
+                httpResponse.close();
```

Comment:
You shouldn't close this always.
In streaming query the connection stays open for the user.
Run the test testPerformanceKustoOperationResultVsJsonVsStreamingQuery to see.